### PR TITLE
Wired members service up to the api and apps!

### PR DESCRIPTION
**Pull request by allouis**
This is the final PR to wire up all of the members stuff, keeping everything behind the labs flag still.

### CODE REVIEW

File: core/test/unit/services/auth/members/index_spec.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const should = require('should');
 const {UnauthorizedError} = require('../../../../../server/lib/common/errors');
 const members = require('../../../../../server/services/auth/members');
 
-describe('Auth Service - Members', function () {
+describe.skip('Auth Service - Members', function () {
```
Comment (allouis) [inline core/test/unit/services/auth/members/index_spec.js]: Skipping because auth service requires members lib and we don't have stubs for the required deps yet!

File: core/server/apps/members/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const membersService = require('../../services/members');
```
Comment (allouis) [inline core/server/apps/members/index.js]: tbh, this feels weird it's so bare - but it was the simplest way to mount the static pages - we might want to reconsider this going forward OR add the ability for "apps" to register apis too

File: core/server/services/auth/members/index.js [hunk 3]
```diff
@@ -1,28 +1,40 @@
...
-    const [scheme, credentials] = req.get('authorization').split(/\s+/);
+module.exports = {
+    get authenticateMembersToken() {
+        if (!labs.isSet('members')) {
+            return function (req, res, next) {
+                return next();
+            };
+        }
+        if (!UNO_MEMBERINO) {
+            UNO_MEMBERINO = jwt({
```
Comment (kirrg001) [inline core/server/services/auth/members/index.js]: lol

File: core/server/apps/members/index.js (hunk 2 above)
Comment (kirrg001) [reply core/server/apps/members/index.js]: Replying here, because i think it fits good.

I tried following the architecture.

It's a little hard to follow why the "members" feature is living in three places now: lib, services, apps.

I guess the only reason is because "apps" can only register frontend/site routes currently?
This probably needs a little re-design in the next phase or if apps are a thing.

Comment (allouis) [reply core/server/apps/members/index.js]: I agree it's a little hard to follow, but I hope we can clear that up going forward - probably with expanding what an "app" can do

#### Why is members in lib?

The members feature was designed to be deployed independently, and eventually be split out into it's own repository, so we would do `require('@tryghost/members-lalala')` - `lib` felt like a good fit for this!

#### Why is members in apps?

You're right, it's in the apps because this was the simplest way to get some frontend/site routes registered.

#### Why is members in service?

Because we're treating `lib/members` as a third party module, we need a service that can give us the shared instance of that module, so we can use it when mounting the app (for the frontend pages) and when mounting the api (for the token, signin etc.. endpoints)

#### How can it be better?

I'm not 100% sure on the future direction, but I hope that apps can become more powerful, and they can register api routes too, this would eliminate the need for the service, and we would have just the `lib/members` which is members specific logic, and then the app, which is what would wire it up with the logic needed for ghost!

If we don't continue with apps as a "thing" then we'll have to come up with something else, maybe a way for services to easily add api and frontend routes, and then we could drop the app stuff.
